### PR TITLE
Mass pings be gone, code owners for all

### PR DESCRIPTION
**Pull request by infinisil**
This PR effectively disables the native GitHub codeowners feature and enables the new alternate codeowners mechanism introduced in https://github.com/NixOS/nixpkgs/pull/336261, meaning that:
- Targeting the wrong branch won't trigger mass pings anymore, and if the wrong base branch is chosen, a [helpful comment](https://github.com/Infinisil-s-Test-Organization/nixpkgs/pull/28#issuecomment-2403617686) is posted telling you how to fix it, so no need to worry about the wrong base branch anymore! (though it might take some time for this to be propagated to non-master branches)
- We can now declare users without write access as code owners!

## Things done

- [x] Made sure that reviews wouldn't be rerequested: https://github.com/NixOS/nixpkgs/pull/347592
- <del>[ ]</del> Checked that it would do the right thing when the wrong base branch is used
  - It's hard to see that because we'd need somebody to wait for ~2 minutes without fixing the problem nor closing a PR, see https://github.com/NixOS/nixpkgs/pull/336261#issuecomment-2403383085
- [x] Checked that codeowners would be requested if enabled: https://github.com/NixOS/nixpkgs/pull/347354#event-14570645380 (that we could see this is a bit of a happy accident, see https://github.com/NixOS/nixpkgs/pull/347538)
- [x] Checked that codeowners problems get detected (https://github.com/NixOS/nixpkgs/pull/347348, https://github.com/NixOS/nixpkgs/pull/346858#issuecomment-2403064093)

---

This work is sponsored by [Antithesis](https://antithesis.com/) :sparkles:

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: .github/OWNERS
Comment (emilazy) [inline .github/OWNERS:115]: This could be @NixOS/acme now :)

**Review by emilazy — COMMENTED** [review]
A shame that we don’t get syntax highlighting any more, I wonder if we could name it `CODEOWNERS` outside of `.github` or something to benefit from the nice colouring (and to avoid stepping on GitHub’s namespace)?

**Review by RossComputerGuy — APPROVED** [review]
LGTM

**Comment by RossComputerGuy** [comment]
> A shame that we don’t get syntax highlighting any more, I wonder if we could name it `CODEOWNERS` outside of `.github` or something to benefit from the nice colouring

Maybe that vim set fmt comment thing might work?

**Comment by infinisil** [comment]
> A shame that we don’t get syntax highlighting any more, I wonder if we could name it `CODEOWNERS` outside of `.github` or something to benefit from the nice colouring (and to avoid stepping on GitHub’s namespace)?

Oh damn, that actually works: https://github.com/Infinisil-s-Test-Organization/nixpkgs/blob/master/ci/CODEOWNERS, though it would also complain about users not having write access in the UI, meh :/

File: .github/OWNERS
Comment (infinisil) [reply .github/OWNERS:115]: I'd like to leave this up to the individual people in follow-up PRs, so that we don't change too much in this one

Comment (emilazy) [reply .github/OWNERS:115]: Yeah for sure, was just happy about it.

**Comment by emilazy** [comment]
I think I’d prefer the warning to no syntax highlighting, unless there’s a way we can get both, but it’s not a big deal. Putting it under `ci/` seems better than `.github/` in any case.

**Comment by emilazy** [comment]
Ah, we can do this: https://github.com/github-linguist/linguist/blob/f164d13fa618023ecf2d8f2ed9a6ce5fae731346/docs/overrides.md

**Comment by emilazy** [comment]
I think if we move it to `ci/OWNERS` and add this to `.gitattributes` then syntax highlighting should work:

```
ci/OWNERS linguist-language= CODEOWNERS
```

**Comment by infinisil** [comment]
I'm not convinced that syntax highlighting at the cost of constant warnings is worth it tbh. Will move to `ci/OWNERS` though.

**Comment by emilazy** [comment]
I suspect we might not get warnings if you set it in `.gitattributes`, but I can’t say for sure.

**Comment by infinisil** [comment]
> I suspect we might not get warnings if you set it in .gitattributes, but I can’t say for sure.

Whoa, that actually [does work](https://github.com/tweag/nixpkgs/blob/codeowners-switch/ci/OWNERS), very nice! I also verified it in [another repo](https://github.com/Infinisil-s-Test-Organization/test/blob/main/foo/OWNERS) because I couldn't believe it at first, but it really does work!

**Review by SigmaSquadron — APPROVED** [review]
LGTM; Thank you for implementing `CODEOWNERS` for the rest of us!

**Comment by emilazy** [comment]
> Whoa, that actually [does work](https://github.com/tweag/nixpkgs/blob/codeowners-switch/ci/OWNERS), very nice! I also verified it in [another repo](https://github.com/Infinisil-s-Test-Organization/test/blob/main/foo/OWNERS) because I couldn't believe it at first, but it really does work!

My intuition was that the component that handles the Linguist‐based syntax highlighting is distinct from the component that checks the syntax and handles displaying the warnings in the UI. Looks like I was right :)

**Review by emilazy — APPROVED** [review]
Let’s try it out. Thanks for tackling this long‐standing problem!

**Comment by infinisil** [comment]
It works! https://github.com/NixOS/nixpkgs/pull/347626

**Comment by jian-lin** [comment]
codeowner check failed for this [PR]: https://github.com/NixOS/nixpkgs/actions/runs/11284183639/job/31384967152?pr=347822

[PR]: https://github.com/NixOS/nixpkgs/pull/347822

**Pull request merged**
